server/controllers/clubProfileController.js

```diff
@@ -82,23 +84,37 @@ module.exports = {
     },
     create: function(req, res) {
         // TODO; req.body contains profile information
-        clubProfile.create(req.body)
-                .then(newclubProfile => res.json(newclubProfile))
+        clubProfile.create(req.body) 
+                .then(newclubProfile => {
+                  //set new profile's acc ID to the given accID
+                  newclubProfile.clubAccountId = req.params.accountId;
+                  //get the account assosiated with given accID and set its profile Id to the new club profile Id
+                  var newClubAccount = getById(req.params.accountId,res);
+                  newClubAccount.clubProfileId = newclubProfile.id;
+                  res.json(newclubProfile)
+                })
                 .catch(err => res.status(422).json(err))
     },
     update: function(req, res) {
         // TODO; req.body contains profile information and req.params.id
-        clubProfile.findOneAndUpdate({ _id: req.params.id}, req.body)
+        // req.params.accountId is account id 
+        // find associated profile from this
+        var updateProfile = getProfile(req.params.accountId,res); 
```

Comment:
same issue with passing res into this function as above; you can simply use the `findOneAndUpdate` function and combine it with the rest of the code too. For example, a sample code block:
```
clubProfile.findOneAndUpdate({clubAccountId: req.params.accountId}, req.body, {new: true})
    .then(clubProfile => res.json(clubProfile))
    .catch(err => res.status(422).json(err))
```